server/src/main/java/org/elasticsearch/search/SearchHit.java

```diff
@@ -635,7 +635,12 @@ public XContentBuilder toInnerXContent(XContentBuilder builder, Params params) t
         } else {
             builder.field(Fields._SCORE, score);
         }
+
         for (DocumentField field : metaFields.values()) {
+            // ignore empty metadata fields
+            if (field.getValues().size() == 0) {
+                continue;
```

Comment:
It would be great to have a a test for this too.

Reply:
++, I pushed https://github.com/elastic/elasticsearch/pull/58418/commits/b947758d96000e4b6aa87b1b4bf531dd1f0f9677